---
title: "Lottia.digitalis"
author: "Lilli Krier"
date: "10-21-19"
output: html_document
---

# Setup

```{r warning=F}
library(rentrez)
library(ape)
library(pegas)
library(strataG)
library(stringr)
library(knitr)
library(ggplot2)
library(reshape2)

#set the working directory
opts_knit$set(root.dir = "~/Users/crandall_lab/Desktop/Lilli_Krier/CapeMendocino")
  


# read in the data to a DNAbin object

Ldigitalis <-read.FASTA("Final_Lottia_Alignments/Lottia_digitalis.fasta")
```

# Initial look at data

Align the data and look at a quick image of it

```{r}
Ldigitalis <- muscle(Ldigitalis)

image.DNAbin(Ldigitalis)

```

## Haplotype network

First make a network that colors each population separately

```{r}
#make a population map by stripping the population name off of the DNAbin labels
pop <- str_extract(rownames(Ldigitalis),"[A-Za-z]+\\s*[A-Za-z]+")

#Label the haplotypes and sort
h <- pegas::haplotype(Ldigitalis)
h <- sort(h, what = "label")
#Create the network
net <- pegas::haploNet(h)
#Funky code to make a table of which haplotypes occur in which populations
i<-stack(setNames(attr(h, "index"), rownames(h)))
i<-i[order(i$values),]
ind.hap<-table(hap=i$ind, pop=pop)
# put columns of ind.hap into North to South order
ind.hap<-ind.hap[,c("Patricks Point","Trinidad Head","Shelter Cove","Kibesillah")]

#Pick some funky colors
mycolors <- c("seagreen1","seagreen3","plum1", "plum3")



plot(net, size=attr(net,"freq"), scale.ratio=2, pie = ind.hap, legend = F, labels = F, threshold = 0, show.mutation = 2, fast = T, bg = mycolors)

legend("bottomleft", colnames(ind.hap), col=mycolors, pch=19, ncol = 2)


```

# Diversity Statistics

```{r}
Ldigitalis_g <- sequence2gtypes(Ldigitalis, strata = pop)

Ldigitalis_g <- labelHaplotypes(Ldigitalis_g)

stratastat<-function(x,pop=pop,fun=nuc.div){
  #this function will calculate stats for a DNAbin object (x), stratified across populations given in pop. 
  # Some functions this will work with: nuc.div(), theta.s(), tajima.test() from pegas, FusFs(), exptdHet() from strataG,
stats<-NULL
for(p in unique(pop)){
  stat<-fun(x[grep(p,rownames(x)),])
  stats<-c(stats,stat)
}
names(stats)<-unique(pop)
return(stats)
}


nucd<-stratastat(Ldigitalis,pop=pop,fun=nuc.div)
Fs<-stratastat(Ldigitalis,pop=pop,fun=fusFs)

kable(data.frame("Nucleotide Diversity"=nucd, "Fu's Fs" = Fs ))

Ldigitalis_g

```



# F-statistics

Took some hints from [here](http://www.sthda.com/english/wiki/ggplot2-quick-correlation-matrix-heatmap-r-software-and-data-visualization)

```{r}

pairwise_phi <- pairwiseTest(Ldigitalis_g$gtypes, stats = "phist", nrep = 10000, model = "raw")

# Let's plot this as a heatmap

  get_lower_tri<-function(x){
    x[upper.tri(x)] <- NA
    return(x)
  }
  
phiST <- pairwise_phi$pair.mat$PHIst

phiST[upper.tri(phiST)] = t(phiST)[upper.tri(phiST)]

phiST <- phiST[c(2,4,3,1),c(2,4,3,1)]
  
phiST[upper.tri(phiST)] <- phiST[lower.tri(phiST)]

melted_phiST <- melt(phiST)
"Patricks Point","Trinidad Head","Shelter Cove","Kibesillah"

levels(melted_phiST$Var1)<-c("Kibesillah", "Shelter Cove", "Trinidad Head", "Patrick's Point")
levels(melted_phiST$Var2)<-c("Kibesillah", "Shelter Cove", "Trinidad Head", "Patrick's Point")
ggplot(data = melted_phiST, aes(Var2, Var1, fill = value)) + geom_tile(color = "white") + scale_fill_gradient(low = "white", high = "orange", name = "phi_ST")



```