---
title: "Nucella ostrina"
author: "Lilli Krier, Eric Crandall"
date: "10-25-19"
output:
  pdf_document:
    toc: yes
  html_document:
    toc: yes
---

# Setup

```{r warning=F, message=F}
library(rentrez)
library(ape)
library(pegas)
library(strataG)
library(stringr)
library(knitr)
library(ggplot2)
library(reshape2)

#set the working directory
#opts_knit$set(root.dir = "~/Users/crandall_lab/Desktop/Lilli_Krier/CapeMendocino")
  


# read in the data to a DNAbin object

Nostrina <-read.FASTA("Final_Ostrina_Alignments/Nucella_ostrina.fasta")

# Initial look at data


Nostrina <- muscle(Nostrina)

image.DNAbin(Nostrina)

```
## Haplotype network

First make a network that colors each population separately

```{r}
#make a population map by stripping the population name off of the DNAbin labels
pop <- str_extract(rownames(Nostrina),"[A-Za-z]+\\s*[A-Za-z]+")

#Label the haplotypes and sort
h <- pegas::haplotype(Nostrina)
h <- sort(h, what = "label")
#Create the network
net <- pegas::haploNet(h)
#Funky code to make a table of which haplotypes occur in which populations
i<-stack(setNames(attr(h, "index"), rownames(h)))
i<-i[order(i$values),]
ind.hap<-table(hap=i$ind, pop=pop)
# put columns of ind.hap into North to South order
ind.hap<-ind.hap[,c("Patricks Point","Trinidad Head","Shelter Cove")]

#Pick some funky colors
mycolors <- c("seagreen1","seagreen3","plum1", "plum3")



plot(net, size=attr(net,"freq"), scale.ratio=2, pie = ind.hap, legend = F, labels = F, threshold = 0, show.mutation = 2, fast = T, bg = mycolors)

#legend("bottomleft", colnames(ind.hap), col=mycolors, pch=19, ncol = 2)


```
# Diversity Statistics

```{r}
Nostrina_g <- sequence2gtypes(Nostrina, strata = pop)

Nostrina_g  <- labelHaplotypes(Nostrina_g)

stratastat<-function(x,pop=pop,fun=nuc.div){
  #this function will calculate stats for a DNAbin object (x), stratified across populations given in pop. 
  # Some functions this will work with: nuc.div(), theta.s(), tajima.test() from pegas, FusFs(), exptdHet() from strataG,
stats<-NULL
for(p in unique(pop)){
  stat<-fun(x[grep(p,rownames(x)),])
  stats<-c(stats,stat)
}
names(stats)<-unique(pop)
return(stats)
}


nucd<-stratastat(Nostrina,pop=pop,fun=nuc.div)
#Fs<-stratastat(Nostrina,pop=pop,fun=fusFs)

nucd
#kable(data.frame("Nucleotide Diversity"=nucd, "Fu's Fs" = Fs ))

Nostrina_g

```

# F-statistics

Took some hints from [here](http://www.sthda.com/english/wiki/ggplot2-quick-correlation-matrix-heatmap-r-software-and-data-visualization), and [here](https://stackoverflow.com/questions/5320814/order-of-rows-in-heatmap)

## Pairwise PhiST

Let's plot the pairwise PhiST statistics as a heatmap

```{r}

pairwise_phi <- pairwiseTest(Nostrina_g$gtypes, stats = "phist", nrep = 10000, model = "raw")



phiST <- pairwise_phi$pair.mat$PHIst

#symmetricize the matrix
phiST[upper.tri(phiST)] = t(phiST)[upper.tri(phiST)]
#put it in North to South Order
phiST<-phiST[c(1,3,2),c(1,3,2)]

distance <- dist(phiST)
cluster <- hclust(distance, method="ward.D")
dendrogram <- as.dendrogram(cluster)
Rowv <- c(3,2,1)
dendrogram <- reorder(dendrogram, Rowv)
reorderfun <- function(d,w) { d }

heatmap(phiST,scale="none",Rowv=dendrogram, reorderfun=reorderfun, cexRow = 1, cexCol=0.8, legend = "col")

```

## AMOVA

```{r}
#write.csv(pop,"amova_hypotheses_lottia.csv",row.names = F, quote=F)
#Open this file up in a spreadsheet program and fill out your hypotheses for higher level hierarchies, with the name of your hypothesis in the header, and pop as the name of the population column

#read them back into R
amovahyps<-read.csv("amova_hypotheses_lottia.csv")
CapeMendocino<-amovahyps$CapeMendocino
pop1<-as.factor(pop)

#Calculate the p-distance among all sequences
dists<-dist.dna(Ldigitalis,model="raw")
```

Test the Cape Mendocino hypothesis:

```{r}
amova_out<-pegas::amova(formula=dists~CapeMendocino/pop1,nperm=1000)

amova_out

# You can see the output above, but we can re-calculate it just to be sure.

FCT<-amova_out$varcomp[1,1]/sum(amova_out$varcomp[,1])
FCTp<-amova_out$varcomp[1,2]
     
FSC<-amova_out$varcomp[2,1]/(amova_out$varcomp[2,1]+amova_out$varcomp[3,1])
FSCp<-amova_out$varcomp[2,2]
      
FST<-(amova_out$varcomp[1,1]+amova_out$varcomp[2,1])/sum(amova_out$varcomp[,1])
FSTp<-NA

result<-c(FCT,FSC,FST)
```


